narwhals/_dask/expr.py

```diff
@@ -71,6 +71,27 @@ def func(df: DaskLazyFrame) -> list[dask_expr.Series]:
             backend_version=backend_version,
         )
 
+    @classmethod
+    def from_column_indices(
+        cls: type[Self],
+        *column_indices: int,
+        backend_version: tuple[int, ...],
+    ) -> Self:
+        def func(df: DaskLazyFrame) -> list[dask_expr.Series]:
+            return [
+                df._native_frame.iloc[:, column_index] for column_index in column_indices
+            ]
+
+        return cls(
+            func,
+            depth=0,
+            function_name="nth",
+            root_names=list(map(str, column_indices)),
+            output_names=list(map(str, column_indices)),
```

Comment:
I think these could just be `None`, which is what we put when we can't keep track of the names because they're not known in advance (like we do with `nw.all`)

`root_names` / `output_names` are use in `group_by`, there's some explanation here: https://narwhals-dev.github.io/narwhals/how_it_works/#group-by

Reply:
@FBruzzesi you mean this? Since we do not know the name, shall we use the column indices? (I can also investigate how Polars did it, but I bet they have access to the column names)

Reply:
Whops, I missed this conversation! My apologies!

I agree that they should not be the column indexes, otherwise _very_ unexpected behaviour would follow.
We should at least add a disclaimer in the function docstring or expand the warning with all the `nw.nth` limitations.